controllers/generic.go

```diff
@@ -0,0 +1,64 @@
+package controllers
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/prest/prest/adapters/postgres"
+)
+
+type HealthCheck struct {
+	Status string `json:"status"`
+}
+
+// Created an interface for future mocking test
+type iDbConnection interface {
+	GetConnection() (*sqlx.DB, error)
+	RunTestQuery() error
+}
+
+type DbConnection struct{}
+
+func (d DbConnection) GetConnection() (db *sqlx.DB, err error) {
+	db, err = postgres.Get()
+	if err != nil {
+		return
+	}
+	return
+}
+
+func (d DbConnection) RunTestQuery() (err error) {
+	db, _ := d.GetConnection()
+	_, err = db.Exec("SELECT 1")
+
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+var dbConn iDbConnection
+
+func init() {
+	dbConn = DbConnection{}
+}
+
+// GetHealthStatus - returns 200 if server is fine, else 424 if Postgres not working
+func GetHealthStatus(w http.ResponseWriter, r *http.Request) {
```

Comment:
A couple things:

1. This func could have a better name, `HealthStatus` would be just fine.
2. I believe the best failure response would be something in the 5xx family, I'd personally use [503](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/503).

Reply:
> 2. I believe the best failure response would be something in the 5xx family, I'd personally use [503](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/503)

`5xx` server error: this problem was not a prestd server error but a client (database) error

> `424` **Failed Dependency:**
> The 424 (Failed Dependency) status code means that the method could not be performed on the resource because the requested action depended on another action and that action failed. For example, if a command in a PROPPATCH method fails, then, at minimum, the rest of the commands will also fail with 424 (Failed Dependency).

In the [Healtch Check **RFC**](https://datatracker.ietf.org/doc/html/draft-inadarei-api-health-check-00) it describes that either `4xx or 5xx` can be used.

> *  "warn": healthy, with some concerns.
> For "pass" and "warn" statuses HTTP response code in the 2xx - 3xx range MUST be used. for "fail" status HTTP response code in the 4xx - 5xx range MUST be used.


I believe `424` fits better

Reply:
> 5xx server error: this problem was not a prestd server error but a client (database) error

the problem with `4xx` client error is that this is also not a client error.

> [15.5. ](https://www.rfc-editor.org/rfc/rfc9110.html#section-15.5)[Client Error 4xx](https://www.rfc-editor.org/rfc/rfc9110.html#name-client-error-4xx)
The 4xx (Client Error) class of status code indicates that the **client seems to have erred**.

I believe that a `424` would be fit if we were calling external services that were dependencies and this request failed due to a bad body or params. With this in mind I'd sugest that we just return a flat `500` as explained bellow, since this is a internal dependency error.

> [15.6.1. ](https://www.rfc-editor.org/rfc/rfc9110.html#section-15.6.1)[500 Internal Server Error](https://www.rfc-editor.org/rfc/rfc9110.html#name-500-internal-server-error)
The 500 (Internal Server Error) status code indicates that the server encountered an unexpected condition that prevented it from fulfilling the request.

Reply:
I will change the controller/function name. 

I chose the 424 status code so we could imply that this was an error occuring on pSQL, not exactly on pREST.

Reply:
As we spoke on twitter, I'm letting the error be a 503 so we can understand that the service is unavailable and it's not necessarily a pREST bad-functioning part.